Polyhedron/demo/Polyhedron/MainWindow.cpp

```diff
@@ -612,8 +616,14 @@ void MainWindow::loadPlugins()
       QString abs_name = pluginsDir.absoluteFilePath(fileName);
       if(loaded.find(abs_name) == loaded.end())
       {
-        load_plugin(abs_name, true);
-        loaded.insert(abs_name);
+        if(load_plugin(abs_name, true))
+        {
+          QString name = QFileInfo(fileName).fileName();
+          name.remove(QRegExp("^lib"));
+          name.remove(QRegExp("\\..*"));
+          PathNames_map[pluginsDir.absolutePath()].push_back(name);
+          loaded.insert(abs_name);
+        }
```

Comment:
If think the lines 
```C++
         QString name = QFileInfo(fileName).fileName();
         name.remove(QRegExp("^lib"));
         name.remove(QRegExp("\\..*"));
         PathNames_map[pluginsDir.absolutePath()].push_back(name);
```
should be in `load_plugin`. That way, if the user loads a plugin with the "File, Load Plugin" menu entry, it will be registered as well.